src/hdmf/backends/hdf5/h5tools.py

```diff
@@ -344,6 +345,33 @@ def read_builder(self):
             self.__read[self.__file] = f_builder
         return f_builder
 
+    def __set_written(self, builder):
+        """
+        Mark this builder as written.
+
+        :param builder: Builder object to be marked as written
+        :type builder: Builder
+        """
+        # currently all values in self._written_builders are True, so this could be a set but is a dict for
+        # future flexibility
+        builder_id = self.__builderhash(builder)
+        self._written_builders[builder_id] = True
+
+    def get_written(self, builder):
+        """Return True if this builder has been written to (or read from) disk by this IO object, False otherwise.
+
+        :param builder: Builder object to get the written flag for
+        :type builder: Builder
+
+        :return: True if the builder is found in self._written_builders using the builder ID, False otherwise
+        """
+        builder_id = self.__builderhash(builder)
+        return self._written_builders.get(builder_id, False)
+
+    def __builderhash(self, obj):
```

Comment:
The same function is implemented here: https://github.com/hdmf-dev/hdmf/blob/3fa0587ae2943aab15a27cce99a17a33883804d8/src/hdmf/build/manager.py#L193

Not sure if it should be generalized, by just adding `__hash__` to Builder. It seems simple enough that its not a big deal to leave as is, but thought I would mention it.

Reply:
Yeah, that's where I got it from. I figured there might have been a reason that it was on the `BuildManager` rather than on `Builder` in the first place, e.g., maybe builders should by default have a hash that is based on its contents, so I left it and re-implemented it here.